reconciler/reconcile_common.go

```diff
@@ -55,4 +60,26 @@ func PostProcessReconcile(ctx context.Context, resource duckv1.KRShaped) {
 	} else if rc.Reason == failedGenerationBump {
 		logger.Warn("A reconciler observed a new generation without updating the resource status")
 	}
+
+	groomConditionsTransitionTime(resource, oldResource)
+}
+
+// groomConditionsTransitionTime ensures that the LastTransitionTime only advances for resources
+// where the condition has changed during reconciliation. This also ensures that all advanced
+// conditions share the same timestamp.
+func groomConditionsTransitionTime(resource, oldResource duckv1.KRShaped) {
```

Comment:
🤔  maybe I'm misunderstanding, but iiuc the entire point of VolatileTime is to avoid what this is doing 😅

Reply:
The VolatileTime strategy has a few weaknesses I've outlined in #1394

- Requires use of k8s specific comparison libraries
- The snapshot of now() is taken multiple times during reconciliation, but all written back at the same time. This can make it hard to tell which conditions are changed in a single reconciliation step (although this is optional, we could keep them).
- It fails to keep the LastTransitionTime stable if a single reconciliation sets conditions more than once (eg. flip-flopping but winds up in the same place).

This obviates the VolatileTime and condition manager strategy so that implementers don't really need to worry about how they are modifying the status struct - it just works.

Reply:
Let me comment on the issue, but it may be worth discussing more synchronous Weds (put on the agenda for API WG?).

Reply:
Yea let's just discuss in WG - that'll be easiest